tests/tests_unit/test_stages.py

```diff
@@ -39,7 +39,7 @@ def test_check_immutable_fields_no_changes(mock_get_state, terraform_state_stage
 def test_check_immutable_fields_mutable_change(
     mock_get_state, terraform_state_stage, mock_config
 ):
-    old_config = mock_config.model_copy()
```

Comment:
flyby: test wasn't failing, but wanted to ensure we were deep copying to prevent unintended errors from a shallow copy